www/class/centreonAuth.SSO.class.php

```diff
@@ -103,7 +102,27 @@ public function __construct(
             # Get configured values
             $clientId = $this->ssoOptions['openid_connect_client_id'];
             $clientSecret = $this->ssoOptions['openid_connect_client_secret'];
-            $redirectNoEncode = $this->ssoOptions['openid_connect_redirect_url'];
+            if (empty($this->ssoOptions['openid_connect_redirect_url'])) {
+                /*
+                 * we can not access $centreon->optGen["oreon_web_path"] before authentication and have to query the
+                 * DB directly
+                 */
+                $pathSql = "SELECT `value` FROM `options` WHERE `key` = 'oreon_web_path'";
```

Comment:
Hi @mayrstefan I think you can remove your SQL query and use $this->ssoOptions['oreon_web_path'];

The CentreonAuthSSO is created with $generalOptions var and this table contains all entries from centreon.options table.
And $this->ssoOptions is a copy of $generalOptions